src/context/FormContext.tsx

```diff
@@ -142,14 +146,32 @@ export const FormProvider = (props: { children: ReactNode }) => {
 
   useEffect(() => {
     if (statusHasChanged) {
+      const currentStatusChangeEvent = loadedTimetableEvents?.find(
+        ({ developmentPlanEvent, endDate }) =>
+          statusChangeEvents.some((e) => e === developmentPlanEvent) && !endDate
+      );
+
       setStatusChangeEvent({
         ...getDefaultTimetableEvent(),
+        ...currentStatusChangeEvent,
         developmentPlan: developmentPlan.reference,
-      });
+        reference: uuidv4(),
+      } as StatusChangeEvent);
     } else {
       setStatusChangeEvent(null);
     }
-  }, [developmentPlan.reference, statusHasChanged]);
+  }, [developmentPlan.reference, loadedTimetableEvents, statusHasChanged]);
+
+  useEffect(() => {
+    const currentStatusChangeFound = loadedTimetableEvents?.some(
+      ({ developmentPlanEvent, endDate }) =>
+        statusChangeEvents.some((e) => e === developmentPlanEvent) && !endDate
+    );
+
+    if (currentStatusChangeFound) {
+      setStatusHasChanged(true);
+    }
+  }, [loadedTimetableEvents]);
```

Comment:
Rather than this being a `useEffect`, I think it would be better to wrap the `setLoadedTimetableEvents` we pass to the context so we don't have to wait for another render when we already have the information we need.

I did have a play around with this, the bit that makes it trickier is the for `setLoadedTimetableEvents` being `Dispatch<SetStateAction<DevelopmentPlanTimetable[] | null>>` instead of `(loadedTimetableEvents: DevelopmentPlanTimetable[] | null) => void`. However, we only ever pass an array or `null`, never a set action, so we could easily update the types.

Reply:
Thanks Ronan :) Done!